tools/gulptasks/lib/test.js

```diff
@@ -17,79 +17,65 @@
 /**
  * Returns list of products affected by modified files staged for commit.
  *
- * @param {Boolean} logPaths
+ * @param {boolean} logPaths
  *        Logging for testing and debugging.
  *
  * @return {Array<string>}
  *         Array of detected products.
  */
 function getProducts(logPaths) {
-    const ChildProcess = require('child_process');
-
-    const paths = ChildProcess
-            .execSync('git diff --cached --name-only --diff-filter=ACM')
-            .toString()
-            .split('\n')
-            .filter(match => !!match),
+    const ChildProcess = require('node:child_process');
+
+    const paths = Array.from(new Set([
+            ...ChildProcess
+                .execSync('git diff --cached --name-only --diff-filter=ACM')
+                .toString()
+                .split('\n')
+                .filter(match => !!match),
+            ...ChildProcess
+                .execSync('git diff refs/heads/master HEAD --name-only --diff-filter=ACM')
```

Comment:
Still see some git weirdness here.
![image](https://github.com/highcharts/highcharts/assets/7478772/0cee8ea2-e5d3-4049-a755-8465c484da36)

Locally this worked for me on a single-branch checkout:
```
➜  shallowcharts git:(tools/product-visual-ci) git remote set-branches origin 'master'
➜  shallowcharts git:(tools/product-visual-ci) git fetch -v --depth=1
➜  shallowcharts git:(tools/product-visual-ci) git diff HEAD origin/master
```